src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -42,11 +42,23 @@ export const fetchMarketData = createAsyncThunk(
 
 export const fetchPriceHistory = createAsyncThunk(
   'marketData/priceHistory',
-  // assets as caip19 array
-  async ({ assets, timeframe }: { assets: string[]; timeframe: HistoryTimeframe }) =>
-    Promise.allSettled(
+  async ({ assets, timeframe }: { assets: CAIP19[]; timeframe: HistoryTimeframe }) => {
+    const responses = await Promise.allSettled(
       assets.map(async asset => await getPriceHistory({ timeframe, ...caip19.fromCAIP19(asset) }))
     )
+
+    const result = responses.reduce<HistoryData[][]>((acc, response) => {
+      if (response.status === 'rejected') return acc
+      const mapped = response.value.map(({ date, price }) => ({
+        date: date.valueOf().toString(), // dates aren't serializable in redux actions or state
```

Comment:
moving this year to the thunk suppresses a browser console warning about actions/state not being serializable in the store. we want to keep the store serializable so we can persist it (or parts of it) through refreshes